Sources/TuistKit/Mappers/Graph/StaticXCFrameworkModuleMapGraphMapper.swift

```diff
@@ -0,0 +1,182 @@
+import Foundation
+import Path
+import TuistCore
+import TuistSupport
+import XcodeGraph
+
+public final class StaticXCFrameworkModuleMapGraphMapper: GraphMapping {
+    private let fileHandler: FileHandling
+
+    public init(
+        fileHandler: FileHandling = FileHandler.shared
+    ) {
+        self.fileHandler = fileHandler
+    }
+
+    public func map(graph: Graph, environment: MapperEnvironment) throws -> (Graph, [SideEffectDescriptor], MapperEnvironment) {
+        let derivedDirectory = derivedDirectory(graph: graph)
+
+        var sideEffects: [SideEffectDescriptor] = []
+        var settingsToPropagate: [GraphDependency: SettingsDictionary] = [:]
+
+        let graphTraverser = GraphTraverser(graph: graph)
+
+        for project in graph.projects.values {
+            for target in project.targets.values {
+                let staticObjcXCFrameworksLinkedByDynamicXCFrameworkDependencies = graphTraverser
+                    .staticObjcXCFrameworksLinkedByDynamicXCFrameworkDependencies(
+                        path: project.path,
+                        name: target.name
+                    )
+                    .compactMap { dependency -> GraphDependency.XCFramework? in
+                        switch dependency {
+                        case let .xcframework(xcframework):
+                            return xcframework
+                        default:
+                            return nil
+                        }
+                    }
+
+                guard !staticObjcXCFrameworksLinkedByDynamicXCFrameworkDependencies.isEmpty else { continue }
+
+                sideEffects += try self.sideEffects(
+                    for: staticObjcXCFrameworksLinkedByDynamicXCFrameworkDependencies,
+                    derivedDirectory: derivedDirectory
+                )
```

Comment:
The lack of semantics in `self.SideEffects` makes it a bit hard what those side effects are all about. What about?

```suggestion
                sideEffects += try self.removeModuleFromHeaderImports(
                    for: staticObjcXCFrameworksLinkedByDynamicXCFrameworkDependencies,
                    derivedDirectory: derivedDirectory
                )
```

Feel free to adjust to anything that feels more accurate to you.